src/actions/entity.js

```diff
@@ -38,3 +38,79 @@ export function updateEntity (entity, componentName, propertyName, value) {
   }
   Events.emit('objectChanged', entity);
 }
+
+/**
+ * Remove an entity
+ * @param  {Element} entity Entity to remove.
+ * @param  {boolean} force (Optional) If true it won't ask for confirmation
+ */
+export function removeEntity (entity, force) {
+  if (entity) {
+    if (force === true || confirm('Do you really want to remove the entity: `' + (entity.id || entity.tagName) + '`')) {
+      entity.parentNode.removeChild(entity);
+      // @todo Select the next entity in the scenegraph
+      editor.selectEntity(null);
+    }
+  }
+}
+
+/**
+ * Remove the selected entity
+ * @param  {boolean} force (Optional) If true it won't ask for confirmation
+ */
+export function removeSelectedEntity (force) {
+  removeEntity(editor.selectedEntity);
```

Comment:
Yep, in fact we have exposed both functions, but on the `EditMenu.js` I used the following code as the menu doesn't have a reference to the current selected entity:
```javascript 
var menuOptions = {
  'Clone': {group: 'base', callback: cloneSelectedEntity, needsEntity: true},
  'Remove': {group: 'base', callback: removeSelectedEntity, needsEntity: true}
};
```
Maybe `callback` could be the one calling `removeEntity(editor..')` and delete the `*selected` on `entity.js`